File: erpnext_gsg/erpnext_gsg/report/attendance_working_hours/attendance_working_hours.py
# Copyright (c) 2023, mahmoud haroun and contributors
# For license information, please see license.txt

import frappe
from frappe.utils import date_diff
from frappe.utils import get_url, sanitize_html




def execute(filters=None):
	columns, data = [], []
	conditions = get_conditions(filters)
	data = get_data(conditions,filters)
	columns = get_columns()
	return columns, data


def get_conditions(filters):
	conditions = ""
	if filters.get("from_date") and filters.get("to_date"):
		conditions += "attendance_date between %(from_date)s and %(to_date)s"

	if filters.get("employee"):
		conditions += " and employee = %(employee)s"

	if filters.get("department"):
		conditions += " and department = %(department)s"

	return conditions

def get_columns():
	columns = [
		{'fieldname': 'employee', 'label':'Employee', 'fieldtype':'Link', 'options':'Employee'},
		{'fieldname': 'employee_name', 'label':'Employee Name', 'fieldtype':'Data'},
		{'fieldname': 'attendance_date', 'label':'Attendance Date', 'fieldtype':'Date'},
		{'fieldname': 'check_in', 'label':'Check In ', 'fieldtype':'Time'},
		{'fieldname': 'check_out', 'label':'Check Out', 'fieldtype':'Time'},
		{'fieldname': 'hours', 'label':'Working Hours', 'fieldtype':'Data'},
		{'fieldname': 'name', 'label':'View Attendance', 'fieldtype':'Data'},
	]
	return columns


def get_data(conditions, filters):
	data = frappe.db.sql(
		"""
		SELECT
			attendance_date, employee ,employee_name, check_in, check_out, hours, name FROM `tabAttendance` WHERE
			{conditions}
	""".format(
			conditions=conditions
		),
		filters,
		as_dict=1,
	)
	for i in data:
		name = i.name
		
		i.name = '<a href="#" onclick="window.open(`http://127.0.0.1:8001/app/attendance/HR-ATT-2023-00001`); return false;" data-doctype="Attendance" data-name="HR-ATT-2023-00001" data-value="HR-ATT-2023-00001" target="_blank">View Attendance</a>'
	return data

File: erpnext_gsg/erpnext_gsg/report/attendance_working_hours/attendance_working_hours.js
// Copyright (c) 2023, mahmoud haroun and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Attendance Working Hours"] = {
	"filters": [
		{
			"fieldname":"from_date",
			"label": __("From Date"),
			"fieldtype": "Date",
			"width": "80",
			"reqd": 1,
			"default": frappe.datetime.add_months(frappe.datetime.get_today(), -1),
		},
		{
			"fieldname":"to_date",
			"label": __("To Date"),
			"fieldtype": "Date",
			"width": "80",
			"reqd": 1,
			"default": frappe.datetime.get_today()
		},
		{
			"fieldname":"employee",
			"label": __("Employee"),
			"fieldtype": "Link",
			"options": "Employee",
			"width": "80",
		},
		{
			"fieldname":"department",
			"label": __("Department"),
			"fieldtype": "Link",
			"options": "Department",
			"width": "80",
		},
		

	]
};
// frappe.templates["my_template_name"] = " \
//     <div class='form-group'> \
//         <label class='control-label' for='my_field_name'>My Field Label</label> \
//         <div class='controls'> \
//             <input type='text' class='form-control' id='my_field_name' name='my_field_name' data-fieldtype='Data' /> \
//         </div> \
//     </div>";
	

frappe.ui.form.on('Attendance Working Hours', {
    refresh: function(frm) {
        // Add HTML attributes to a link in the report
		print("*"*100)
        frm.fields_dict['name'].get_query = function(doc) {
            return {
                filters: {
                    'doctype': 'Attendance',
                    'fieldname': 'name',
                    'link': '<a href="/app/attendance/HR-ATT-2023-00001" data-doctype="Attendance" data-name="HR-ATT-2023-00001" data-value="HR-ATT-2023-00001" target="_blank>HR-ATT-2023-00001</a>'
					
                }
            };
        };
    }
});